### ISSUE TITLE
parameterization

### ISSUE THREAD

**Issue by fengfeng-xie** [sdust-mmlab/SAR_CD_CL#2]
In main_pseudo_label_v1.py,I'd like to know how these parameters are set：T1、step、T2 and af

**Comment by fangs99** [comment]
Dear Xie:

Thanks for your question.

As to these parameters, step refers to the current iteration number in the loop, which is acquired from the main loop. It needs no setup.

The T1, T2 and af are three hyper-parameters, which are set according to experience.

Before T1,  the model is trained using the  (pseudo) labeled data completely,  and the value is related to iteration numbers you believe it is enough for training using the pseudo-labeled data, while during from T1 to T2  the unlabeled data is gradually integrated into training, and the value relates to the ratio of the unlabeled data and how much you hope the unlabeled data affect the model.  Taking into account the unlabeled data is far more than labeled data， we set T1 as 100 and T2 as 600. Of course, the total iteration numbers should be calculated when you set these parameters. Based on our experience, the gap between T1 and T2 is decided by the importance placed on the unlabeled data and its amount.

As to af, we set it as 3 according to the reference Pseudo-label: The simple and efficient semi-supervised learning method for deep neural networks, as stated in our paper. 

In fact, although these hyper-parameters have effect on the results, the effect is not too much. Other values in our test is also acceptable. 

Hope this reply is helpful for you.




--

Prof. Fang Sheng,
Room 306, Building 13,
College of Information Science & Engineering 
Shandong University of Science & Technology, 
Qingdao 266510, P.R.China
***@***.***
      ***@***.***




At 2025-05-09 18:13:19, "fengfeng-xie" ***@***.***> wrote:

fengfeng-xie created an issue (sdust-mmlab/SAR_CD_CL#2)

In main_pseudo_label_v1.py,I'd like to know how these parameters are set：T1、step、T2 and af

**Comment by fengfeng-xie** [comment]
Thank you for taking the time out of your busy schedule to reply to me, and I hope your research goes to the next level!

**Comment by fengfeng-xie** [comment]
Hello author, I apologize for bothering you again. The problem is also about the semisup_train part in that last code file, at first my model was decreasing in the train_supervised part of the train_loss and test_loss, but then in the semisup_train part I noticed that the test_loss would keep on going up and exceeding 1 especially especially when step>T2. I would like to know how to solve this problem

Translated with DeepL.com (free version)

**Comment by fengfeng-xie** [comment]
Hello author, I apologize for bothering you again. The problem is also about the semisup_train part in that last code file, at first my model was decreasing in the train_supervised part of the train_loss and test_loss, but then in the semisup_train part I noticed that the test_loss would keep on going up and exceeding 1 especially especially when step>T2. I would like to know how to solve this problem

**Comment by fangs99** [comment]
Dear Xie,
Thanks for your question. The problem may be generated due to many factors.
Acoording to your description, if step > T2, it means that all unlabeled data has been integrated into the  trianing process. If the test loss becomes increasing gradualy, I guess it maybe resutled from two possible reasons. 
1. Your model mayb be overfitted, so you can use the stop early trick.
2. Check your dataset to make sure your data distribution satisfying the semi-supervised learning requriment. Please notice that the semi-uspervised tech. is valid only when the distributions keep consistent between labeled and unlabeled data. In fact, we are doing someworks to deal the quesiton of the in-consistency among labeled and unlabeled data. Any tech. is valid only when its condition is satisfied, so you should be acquantied your dataset charicteristics. 




Hope these advices are helpful for you.

--

Dr. Fang Sheng,
Room 306, Building 13,
College of Information Science & Engineering 
Shandong University of Science & Technology, 
Qingdao 266510, P.R.China
***@***.***
      ***@***.***




At 2025-05-13 10:08:07, "fengfeng-xie" ***@***.***> wrote:

fengfeng-xie left a comment (sdust-mmlab/SAR_CD_CL#2)

Hello author, I apologize for bothering you again. The problem is also about the semisup_train part in that last code file, at first my model was decreasing in the train_supervised part of the train_loss and test_loss, but then in the semisup_train part I noticed that the test_loss would keep on going up and exceeding 1 especially especially when step>T2. I would like to know how to solve this problem

**Comment by fengfeng-xie** [comment]
Thanks for your reply, I'll look into it more carefully, thanks!

**Comment by fengfeng-xie** [comment]
Hello author, I hope you can understand that I have more questions because I want to join this semi-supervisory mechanism of yours. The total number of uncertain batch in my dataset is 1277, and I set the unlabel_interval to 50, is it because the supervised bootstrapping is too little so that is the cause. And I found that especially at the end when we give the highest weight to the results of the uncertain samples, the testloss will be higher!

**Comment by fangs99** [comment]
Dear Xie,

Thanks for your question.

My suggestions are the following :

1.  Check your the distribution of your dataset , especilly the labeled and unlabeled data,.

2. "especially at the end when we give the highest weight to the results of the uncertain samples, the testloss will be higher", the af can be decreased to be less, for example 2, or 1.

3.  During training, the labeled data can be exploied (used for training) more frequently than now, in order to keep the model not  to depart  too much due to the effect of the unlabeled data.

I hope these will be helpful for you!

Best regards!

--

Dr. Fang Sheng,
Room 306, Building 13,
College of Information Science & Engineering 
Shandong University of Science & Technology, 
Qingdao 266510, P.R.China
***@***.***
      ***@***.***




At 2025-05-13 18:46:29, "fengfeng-xie" ***@***.***> wrote:

fengfeng-xie left a comment (sdust-mmlab/SAR_CD_CL#2)

Hello author, I hope you can understand that I have more questions because I want to join this semi-supervisory mechanism of yours. The total number of uncertain batch in my dataset is 1277, and I set the unlabel_interval to 50, is it because the supervised bootstrapping is too little so that is the cause. And I found that especially at the end when we give the highest weight to the results of the uncertain samples, the testloss will be higher!
